### PR TITLE
xcache agis-to-rucio probe

### PR DESCRIPTION
**Pull request by mlassnig**

### CODE REVIEW

File: common/check_xcache [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+            for se in SE_DATA[rc_site]:
+
+                for protocol in SE_DATA[rc_site][se]['protocols']:
+
+                    # find all active internal proxies (==xcaches)
+                    if SE_DATA[rc_site][se]['protocols'][protocol]['doortype'] == 'proxyinternal' and\
+                       SE_DATA[rc_site][se]['protocols'][protocol]['state'] == 'ACTIVE':
+                        xcaches.append( (rc_site, SE_DATA[rc_site][se]['protocols'][protocol]['endpoint']) )
+
+        for xcache in xcaches:
```
Comment (dchristidis) [inline common/check_xcache]: I would suggest to implement the reverse flow too. If the probe is put into production, users will expect removing a protocol from AGIS to have an effect in Rucio.

Comment (mlassnig) [reply common/check_xcache]: aye aye captain!

File: common/check_xcache [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+if __name__ == '__main__':
+
+    try:
+
+        session = get_session()
+
+        xcaches_in_rucio = {}
+        res = core_items('root-proxy-internal', session=session)
+        for site,host in res:
```
Comment (dchristidis) [inline common/check_xcache]:
```
p:59:16: C0326: Exactly one space required after comma
        for site,host in res:
                ^ (bad-whitespace)
```

File: common/check_xcache [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+                    if SE_DATA[rc_site][se]['protocols'][protocol]['doortype'] == 'proxyinternal' and\
+                       SE_DATA[rc_site][se]['protocols'][protocol]['state'] == 'ACTIVE':
+                        xcaches_in_agis.append( (rc_site, SE_DATA[rc_site][se]['protocols'][protocol]['endpoint']) )
+
+        print('Setting values in Rucio from AGIS:')
+        for xcache in xcaches_in_agis:
+            url = '%s:%s' % (urlparse(xcache[1]).hostname, urlparse(xcache[1]).port)
+            print('-SET->', xcache[0], '--->', url)
+            core_set('root-proxy-internal', xcache[0], url, session=session)
+            del(xcaches_in_rucio[xcache[0]])
```
Comment (dchristidis) [inline common/check_xcache]:
```
p:88:0: C0325: Unnecessary parens after 'del' keyword (superfluous-parens)
```

File: common/check_xcache [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        for rc_site in SE_DATA:
+
+            for se in SE_DATA[rc_site]:
+
+                for protocol in SE_DATA[rc_site][se]['protocols']:
+
+                    # find all active internal proxies (==xcaches_in_agis)
+                    if SE_DATA[rc_site][se]['protocols'][protocol]['doortype'] == 'proxyinternal' and\
+                       SE_DATA[rc_site][se]['protocols'][protocol]['state'] == 'ACTIVE':
+                        xcaches_in_agis.append( (rc_site, SE_DATA[rc_site][se]['protocols'][protocol]['endpoint']) )
```
Comment (dchristidis) [inline common/check_xcache]:
```
p:81:46: C0326: No space allowed after bracket
                        xcaches_in_agis.append( (rc_site, SE_DATA[rc_site][se]['protocols'][protocol]['endpoint']) )
                                              ^ (bad-whitespace)
p:81:115: C0326: No space allowed before bracket
                        xcaches_in_agis.append( (rc_site, SE_DATA[rc_site][se]['protocols'][protocol]['endpoint']) )
                                                                                                                   ^ (bad-whitespace)
```

File: common/check_xcache [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+                    if SE_DATA[rc_site][se]['protocols'][protocol]['doortype'] == 'proxyinternal' and\
+                       SE_DATA[rc_site][se]['protocols'][protocol]['state'] == 'ACTIVE':
+                        xcaches_in_agis.append((rc_site, SE_DATA[rc_site][se]['protocols'][protocol]['endpoint']))
+
+        print('Setting values in Rucio from AGIS:')
+        for xcache in xcaches_in_agis:
+            url = '%s:%s' % (urlparse(xcache[1]).hostname, urlparse(xcache[1]).port)
+            print('-SET->', xcache[0], '--->', url)
+            core_set('root-proxy-internal', xcache[0], url, session=session)
+            del xcaches_in_rucio[xcache[0]]
```
Comment (dchristidis) [inline common/check_xcache]: Sorry Mario, this raises a `KeyError` when a cache is added for the first time.

**Pull request merged**